core/src/engine/sleep.cpp

```diff
@@ -23,8 +23,10 @@ class CommonSleepWaitStrategy final : public WaitStrategy {
 
 void InterruptibleSleepUntil(Deadline deadline) {
   auto& current = current_task::GetCurrentTaskContext();
+  current.SetBackground(true);
   impl::CommonSleepWaitStrategy wait_manager{};
   current.Sleep(wait_manager, deadline);
+  current.SetBackground(false);
```

Comment:
Use `FastScopeGuard` to ensure that if `SetBackground(true)` is called, then `SetBackground(false)` will surely be called

`Sleep` should not normally throw, but just in case